internal/provider/nordvpn/api/server.go

```diff
@@ -0,0 +1,32 @@
+package api
+
+import (
+	"strconv"
+	"strings"
+)
+
+type server struct {
+	Groups       []group
+	Hostname     string
+	Ips          []ip
+	Load         int
+	Locations    []location
+	Name         string
+	Status       string
+	Technologies []technology
+}
+
+func (s *server) number() uint16 {
+	i := strings.IndexRune(s.Name, '#')
+	if i < 0 {
+		return 0
+	}
+
+	idString := s.Name[i+1:]
+	idUint64, err := strconv.ParseUint(idString, 10, 16) //nolint:gomnd
+	if err != nil {
+		return 0
```

Comment:
this is not valid, if the number cannot be parsed, an error should be issued and we should not silently discard it and return `0` instead. Please use the older removed function `parseServerName` for this purpose.